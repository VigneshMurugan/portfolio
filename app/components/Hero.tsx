import classnames from "classnames";
import { ReactElement } from "react";

export interface HeroProps {
  className?: string;
  copy?: string | ReactElement;
  highlight: string;
}

export const Hero = (props: HeroProps) => {
  const { className, copy, highlight } = props;

  const cssComponent = classnames("hero p-4 text-center", className);

  return (
    <div className={cssComponent}>
      <h1 className="font-font-serif text-4xl font-extrabold md:text-4xl">
        {copy && copy}
        <div className="text-highlight text-5xl md:text-7xl">{highlight}</div>
      </h1>
    </div>
  );
};
