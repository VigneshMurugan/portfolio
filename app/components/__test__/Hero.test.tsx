import { expect, test } from "vitest";
import { render } from "@testing-library/react";

import { Hero } from "../Hero";

describe("Hero Component", () => {
  const component = render(<Hero copy="Test Copy" highlight="The HIGHLIGHT" />);

  test("should render our Hero", () => {
    expect(component.baseElement).toMatchSnapshot();
  });
});
