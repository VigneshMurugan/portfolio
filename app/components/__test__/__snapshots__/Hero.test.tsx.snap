// Vitest Snapshot v1

exports[`Hero Component > should render our Hero 1`] = `
<body>
  <div>
    <div
      class="hero p-4 text-center"
    >
      <h1
        class="font-font-serif text-4xl font-extrabold md:text-4xl"
      >
        <div>
          Test Copy
        </div>
        <div
          class="text-highlight text-5xl md:text-7xl"
        >
          The HIGHLIGHT
        </div>
      </h1>
    </div>
  </div>
</body>
`;
